src/GraphQL/Types/Composite/ComplexGraphType.cs

```diff
@@ -559,6 +559,7 @@ public virtual FieldType AddField(FieldType fieldType)
         /// <param name="expression">The property of the source object represented within an expression.</param>
         /// <param name="nullable">Indicates if this field should be nullable or not. Ignored when <paramref name="type"/> is specified.</param>
         /// <param name="type">The graph type of the field; if <see langword="null"/> then will be inferred from the specified expression via registered schema mappings.</param>
+        [Obsolete("Please use another overload that receives only one of the 'nullable' or 'type' arguments. This method will be removed in v9.")]
```

Comment:
I suggest making it private in v9 instead of removing it so there won't be code duplicates between other overloads.